### ISSUE TITLE
How about intoducing auto-roll for output window

### ISSUE THREAD

**Issue by idealkindom** [Proektsoftbg/Calcpad#118]
When a calculation sheet is long, especially when there is #pause and #input, the auto-roll of output window could be very helpful.

**Comment by Proektsoftbg** [comment]
Do you mean to scroll automatically to the bottom? Can you please specify what does "auto-roll" mean and how it is supposed to work? I am not quite sure about it.

**Comment by idealkindom** [comment]
> Do you mean to scroll automatically to the bottom? Can you please specify what does "auto-roll" mean and how it is supposed to work? I am not quite sure about it.
Yes,.scroll to when it pause or stop calculation.

**Comment by idealkindom** [comment]
> Do you mean to scroll automatically to the bottom? Can you please specify what does "auto-roll" mean and how it is supposed to work? I am not quite sure about it.

Yes, scroll to where the calculation pause or stop.

**Comment by Proektsoftbg** [comment]
Yes, it is possible, but what if the user wants to review the results from the beginning. Then he/she will need to scroll back each time. We discussed this some time ago and decided that the best is to preserve the scroll at the same place after each calculation. I work like this from a while and it is very convenient for most cases.

Probably, if any automatic scroll is introduced, it should be selectable by a checkbox.

**Comment by idealkindom** [comment]
> d, it should be selectable by a checkbox.

Since #pause and #input are incurred now, I think it would be more convenient to auto-scroll to the recent output to interact with the calculation. Maybe it's advisable to make this selective for the users or let it be triggered by some keywords, i.e. #pause and #input,etc.

**Comment by idealkindom** [comment]
> Yes, it is possible, but what if the user wants to review the results from the beginning. Then he/she will need to scroll back each time. We discussed this some time ago and decided that the best is to preserve the scroll at the same place after each calculation. I work like this from a while and it is very convenient for most cases.
> 
> Probably, if any automatic scroll is introduced, it should be selectable by a checkbox.

Also, it there are some errors in the source code, it would be instant to find the errors, this may call the output window and the source window to scroll simutaneously.

**Comment by Proektsoftbg** [comment]
OK, maybe on #pause and #input would be more convenient to scroll down to the end (the breakpoint) but I need to play a little bit more to see how it feels. :)

Thank you for your suggestion! :)

**Comment by idealkindom** [comment]
> OK, maybe on #pause and #input would be more convenient to scroll down to the end (the breakpoint) but I need to play a little bit more to see how it feels. :)
> 
> Thank you for your suggestion! :)

BTW,I think to see the output.and sourcecode window content to scroll is a cool feeling.
